src/cose/test_cose.rs

```diff
@@ -22,43 +23,115 @@ const P256_PARAMS: SignatureParameters = SignatureParameters {
     algorithm: SignatureAlgorithm::ES256,
     pkcs8: &test::PKCS8_P256_EE,
 };
+const P384_PARAMS: SignatureParameters = SignatureParameters {
+    certificate: &test::P384_EE,
+    algorithm: SignatureAlgorithm::ES384,
+    pkcs8: &test::PKCS8_P384_EE,
+};
+const P521_PARAMS: SignatureParameters = SignatureParameters {
+    certificate: &test::P521_EE,
+    algorithm: SignatureAlgorithm::ES521,
+    pkcs8: &test::PKCS8_P521_EE,
+};
 
-#[test]
-fn test_cose_sign_verify() {
+#[cfg(test)]
+fn test_verify(payload: &[u8], cert_chain: &[&[u8]], params_vec: Vec<SignatureParameters>) {
     test::setup();
-    let payload = b"This is the content.";
-    let certs: [&[u8]; 2] = [&test::P256_ROOT,
-                             &test::P256_INT];
-    let params_vec = vec![P256_PARAMS];
-    let cose_signature = sign(payload, &certs, &params_vec);
+    let cose_signature = sign(payload, &cert_chain, &params_vec);
```

Comment:
Do we need to take a reference to the cert_chain here? It's already a reference...

Reply:
We don't need another reference here. I would've thought the compiler catches this...